clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreBufferService.java

```diff
@@ -369,16 +390,14 @@ public void putConsumerRecord(
       int subPartition,
       String kafkaUrl,
       long beforeProcessingRecordTimestamp) throws InterruptedException {
-    int drainerIndex = getDrainerIndexForConsumerRecord(consumerRecord, subPartition);
-    MemoryBoundBlockingQueue<QueueNode> queue = blockingQueueArr.get(drainerIndex);
     if (leaderProducedRecordContext == null) {
       /**
        * The last queued record persisted future will only be setup when {@param leaderProducedRecordContext} is 'null',
        * since {@link LeaderProducedRecordContext#persistedToDBFuture} is a superset of this, which is tracking the
        * end-to-end completeness when producing to local Kafka is needed.
        */
       CompletableFuture<Void> recordFuture = new CompletableFuture<>();
-      queue.put(
+      getDrainerForConsumerRecord(consumerRecord, subPartition).put(
```

Comment:
Question here... is it right to put something in the queue, and then, after that, call `partitionConsumptionState.setLastQueuedRecordPersistedFuture`? Is there a race condition there?